### PR TITLE
[13/N] Fix extra warnings brought by clang-tidy-17 

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/core/ivalue.cpp [hunk 1]
```diff
@@ -574,11 +574,9 @@ static std::ostream& printMaybeAnnotatedDict(
 static std::ostream& printComplex(std::ostream & out, const IValue & v) {
   c10::complex<double> d = v.toComplexDouble();
   IValue real(d.real()), imag(std::abs(d.imag()));
-  auto sign = "";
+  auto sign = '-';
```
Comment (Skylion007) [inline aten/src/ATen/core/ivalue.cpp:577]: one of the few times this actually might be cleaner as a single line ternary?

File: torch/csrc/distributed/autograd/context/context.cpp [hunk 2]
```diff
@@ -115,7 +115,7 @@ std::shared_ptr<torch::autograd::GraphTask> DistAutogradContext::
 }
 
 void DistAutogradContext::setGraphTask(
-    std::shared_ptr<torch::autograd::GraphTask> graphTask) {
+    const std::shared_ptr<torch::autograd::GraphTask>& graphTask) {
```
Comment (Skylion007) [inline torch/csrc/distributed/autograd/context/context.cpp:118]: Now the std::move in the function is useless though? Why was this changed?

File: torch/csrc/autograd/variable.cpp [hunk 3]
```diff
@@ -269,7 +269,7 @@ void create_cpp_hook(const at::TensorBase& self, bool is_retains_grad_hook) {
 
 void set_grad_accumulator(
     const Variable& self,
-    std::weak_ptr<Node> grad_accumulator) {
+    const std::weak_ptr<Node>& grad_accumulator) {
```
Comment (Skylion007) [inline torch/csrc/autograd/variable.cpp:272]: Also now the move is pointless?

File: aten/src/ATen/cuda/tunable/StreamTimer.cpp [hunk 4]
```diff
@@ -34,7 +34,7 @@ void StreamTimer::End() {
 }
 
 float StreamTimer::Duration() {
-  float time;
+  float time = NAN;
```
Comment (Skylion007) [inline aten/src/ATen/cuda/tunable/StreamTimer.cpp:37]:
```suggestion
  auto time = std::numeric_limits<float>::quiet_NaN()
```

File: aten/src/ATen/MatrixRef.h [hunk 5]
```diff
@@ -92,6 +92,7 @@ class MatrixRef {
   /// The declaration here is extra complicated so that "arrayRef = {}"
   /// continues to select the move assignment operator.
   template <typename U>
+  // NOLINTNEXTLINE(cppcoreguidelines-missing-std-forward)
```
[the remark this answers is not in the document]
Comment (cyyever) [reply aten/src/ATen/MatrixRef.h:95]: Newer versions have better detection while it seems like a false positive.

File: torch/csrc/distributed/autograd/context/context.cpp (hunk 2 above)
Comment (cyyever) [reply torch/csrc/distributed/autograd/context/context.cpp:118]: Reverted

File: aten/src/ATen/core/ivalue.cpp (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/core/ivalue.cpp]: Fixed

File: torch/csrc/autograd/variable.cpp (hunk 3 above)
Comment (cyyever) [reply torch/csrc/autograd/variable.cpp:272]: Reverted

Comment (Skylion007) [reply torch/csrc/autograd/variable.cpp:272]: Doesn't look reverted to me?

Comment (cyyever) [reply torch/csrc/autograd/variable.cpp]: Had mistakes, now it is reverted

**Review by ezyang — APPROVED** [review]
